target/linux/ramips/mt7621/base-files/lib/upgrade/linksys.sh

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright (C) 2014 OpenWrt.org
+#
+
+linksys_get_target_firmware() {
+
+	local cur_boot_part mtd_ubi0
+
+	cur_boot_part=$(/usr/sbin/fw_printenv -n boot_part)
+	if [ -z "${cur_boot_part}" ] ; then
+		mtd_ubi0=$(cat /sys/devices/virtual/ubi/ubi0/mtd_num)
+		case $(egrep ^mtd${mtd_ubi0}: /proc/mtd | cut -d '"' -f 2) in
+		kernel|rootfs)
+			cur_boot_part=1
+			;;
+		alt_kernel|alt_rootfs)
+			cur_boot_part=2
+			;;
+		esac
+		>&2 printf "Current boot_part='%s' selected from ubi0/mtd_num='%s'" \
+			"${cur_boot_part}" "${mtd_ubi0}"
+	fi
+
+	case $cur_boot_part in
+	1)
+		fw_setenv -s - <<-EOF
+			boot_part 2
+			bootcmd "run altnandboot"
+		EOF
+		printf "kernel2"
+		return
+		;;
+	2)
+		fw_setenv -s - <<-EOF
+			boot_part 1
+			bootcmd "run nandboot"
+		EOF
+		printf "kernel1"
+		return
+		;;
+	*)
+		return
+		;;
```

Comment:
This default case is useless and should be removed.

Reply:
Sure, NP. I admit, copied linksys.sh from another target - but agree with you, updating it!